File: src/java/magmac/app/lang/node/Continue.java
package magmac.app.lang.node;

import magmac.api.Option;
import magmac.app.compile.error.CompileResult;
import magmac.app.compile.node.MapNode;
import magmac.app.compile.node.Node;
import magmac.app.lang.Destructors;
import magmac.app.lang.java.JavaFunctionSegmentValue;
import magmac.app.lang.java.Lang;

class Continue implements JavaFunctionSegmentValue, Lang.TypescriptFunctionSegmentValue {
    public static Option<CompileResult<Continue>> deserialize(Node node) {
        return Destructors.destructWithType("continue", node).map(deserializer -> deserializer.complete(Continue::new));
    }

    @Override
    public Node serialize() {
        return new MapNode("continue");
    }
}

File: src/web/magmac/app/lang/node/Continue.ts
import { Option } from "../../../../magmac/api/Option";
import { CompileResult } from "../../../../magmac/app/compile/error/CompileResult";
import { MapNode } from "../../../../magmac/app/compile/node/MapNode";
import { Node } from "../../../../magmac/app/compile/node/Node";
import { Destructors } from "../../../../magmac/app/lang/Destructors";
import { JavaFunctionSegmentValue } from "../../../../magmac/app/lang/java/JavaFunctionSegmentValue";
import { Lang } from "../../../../magmac/app/lang/java/Lang";
export class Continue {
	deserialize(node : Node) : Option<CompileResult<Continue>> {;;}
	serialize() : Node {;;}
}
